pkg/logfilter/logfilter.go

```diff
@@ -0,0 +1,77 @@
+package logfilter
+
+import (
+	"net/url"
+	"strings"
+)
+
+type Filter interface {
+	Allow(string)
+	IsAllowed(string) bool
+	FilterURL(url *url.URL) *url.URL
+	FilterURLString(url string) string
+	FilterQueryParams(params url.Values) url.Values
+}
+
+type mapKey struct{}
+
+type logFilter struct {
+	allowedMap map[string]mapKey
+}
+
+var _ Filter = &logFilter{}
+
+func New() Filter {
+	return &logFilter{allowedMap: make(map[string]mapKey)}
+}
+
+func (l *logFilter) Allow(key string) {
+	l.allowedMap[strings.ToLower(key)] = mapKey{}
+}
+
+func (l *logFilter) IsAllowed(key string) bool {
+	if _, ok := l.allowedMap[strings.ToLower(key)]; ok {
+		return true
+	}
+
+	return false
+}
+
+func (l *logFilter) FilterURLString(urlString string) string {
+	parsedUrl, err := url.Parse(urlString)
+
+	if err != nil {
+		return urlString
+	}
+
+	return l.FilterURL(parsedUrl).String()
+}
+
+func (l *logFilter) FilterURL(originalUrl *url.URL) *url.URL {
+	clonedUrl := *originalUrl
+
+	if clonedUrl.User != nil {
+		clonedUrl.User = url.UserPassword("FILTERED", "FILTERED")
+	}
+
+	filteredParams := l.FilterQueryParams(clonedUrl.Query())
+	clonedUrl.RawQuery = filteredParams.Encode()
+
+	return &clonedUrl
+}
+
+func (l *logFilter) FilterQueryParams(query url.Values) url.Values {
+	filteredQueryParams := make(url.Values)
```

Comment:
Might be worth sizing this initially to be the same size as `query`

```suggestion
	filteredQueryParams := make(url.Values, 0, len(query))
```

Reply:
Nevermind, that doesn't work with `url.Values`.

Reply:
Actually, sorry, I think this would work:

```suggestion
	filteredQueryParams := make(url.Values, len(query))
```

Reply:
It looks like you can give it a hint of the size (but not enforce bounds) via `filteredQueryParams := make(url.Values, len(query))`